### ISSUE TITLE
nearest_neighbors method is leasing db connection and not releasing

### ISSUE THREAD

**Issue by uznadeem** [ankane/neighbor#26]
Following code in our project is leasing the db connection but not releasing.

```
candidates.nearest_neighbors(:embedding, search_embedding, distance: 'cosine')
```

As a result, we are needed to increase our db pool in order to manage the concurrency load which would definitely increase the cost.

For now we are forcefully releasing the db connection wherever we are using the above line.

```
ActiveRecord::Base.connection_pool.release_connection
```

It would be great if we release the db connect `nearest_neighbors` method.

**Comment by ankane** [comment]
Hi @uznadeem, it'll need to check out a connection to run the query, but it shouldn't be doing any manual connection handling.

**Comment by ankane** [comment]
Looking into this more, it seems to have been caused by a breaking change in Active Record 7.2 (https://github.com/rails/rails/pull/51192). Will figure out a fix.

**Comment by ankane** [comment]
Fixed in the commit above. Thanks for reporting!

**Comment by Petercopter** [comment]
OK so funny story. I upgraded to Rails 7.2 a few weeks ago, and then started playing with the neighbor gem. I implemented it into an autocomplete search without debouncing just to test it. Oh and no index on the embeddings either, lol.

It completely brought the Heroku dyno to it's knees. I kept upgrading the dyno and I finally got it stable, but it was never releasing the memory.

Between debouncing, database index and this fix, things work great! Thank you so much for the fix, and for the gem. Postgres ftw!
